### ISSUE TITLE
Add "Add to cart" api endpoint

### ISSUE THREAD

**Issue by valincius** [Code2Gether-Discord/Congo#87]
Need to add a endpoint to the cart controller that takes a product id & quantity and adds to our cart

**Comment by valincius** [comment]
Might need more of a discussion but I think here is where we should also build the cart record if one isn't passed in.
We can have a cookie with the cart id that is passed with the "add to cart" request, if we don't pass one in, we can have this action create the cart then return the id then can use that for subsequent requests?

**Comment by pobiega** [comment]
That's the kind of flow we want to enable, and a cookie with cart ID seems like a good solution. So just to be clear, the API endpoint will either respond with **204 No Content** or **200 OK** with a cartID as the response body, assuming happy flow. Is that ok with you guys?

**Comment by Palisar** [comment]
Sounds good i have no idea how cookies work but i guess now is as good a time as any

**Comment by valincius** [comment]
I think we may be able to avoid the round trip and just have the client generate the cart id since it's a guid.
Then the server won't have to send anything back. It can just create the cart record if it's a cart id that doesn't exist in the system, then add the item to it.

**Comment by Palisar** [comment]
yeah and this action would only happen once the first item is added right ? after that would we not want the cookie to store the guid ? 
FYI i may have to wait until the weekend to finish this properly, im running on very little sleep today

**Comment by Palisar** [comment]
Gonna setup a PR draft for this  as I think its going to require a few steps to complete

**Comment by valincius** [comment]
> yeah and this action would only happen once the first item is added right ? after that would we not want the cookie to store the guid ?
> FYI i may have to wait until the weekend to finish this properly, im running on very little sleep today

We would still want to store the cart id, we’d just already have it before sending the request if we go with this route. You can go with whatever approach though, whichever you think is easier!

**Comment by Palisar** [comment]
This task is complete I forgot to add the PR to the issue
